### ISSUE TITLE
Screenshot of progress?

### ISSUE THREAD

**Issue by ghost** [tstr/TSEngine#2]
I love finding these, but I never have the time to clone and compile just to see a preview. Any chance a screenshot is planned?

**Comment by BrotherhoodOfHam** [comment]
Sure I can take a few screenshots I've been meaning to for a long time. I think some of the older versions of this repo actually look better than the current version in my opinion I should really move them to a different repository.

**Comment by BrotherhoodOfHam** [comment]
If you're still interested [I forked the old version of this repo](https://github.com/BrotherhoodOfHam/SponzaRenderer) which is more complete from a user perspective compared to this. There are some screenshots and you can download the compiled binaries if you choose.

I'll maybe have screenshots for this version one day :)

**Comment by communityus** [comment]
Wow, it looks really good! I have a few other questions if you have time. What is best way to ask them? Do you have a Steam account perchance? Since this is a game related library might make sense to chat there. Esp. Interested in what's diff in new code, and a few platform specific questions. Also curious how much more do you plan to do on this work.

**Comment by BrotherhoodOfHam** [comment]
Sure ask away. We could use steam if you want or something like discord (discord might be preferable). I use the same username as on github.
I do plan to do more work but it all depends on how much time I have. It's not really a serious library, it's more of an educational excersize. Talking about it is always interesting though.

**Comment by BrotherhoodOfHam** [comment]
I should say any more work I do on this project will likely be less game-centric and more about the renderer. Afterwards I'll start thinking about other systems.

**Comment by communityus** [comment]
Okay, I'll ask the first few questions here. 

Have you at all looked at playing with the Vulkan spec, I see DX11 now?

Above question of course leads us to other non windows platforms, have any design decisions been made along the way with future cross platform ease of use?

The modularity of the engine looks like it begins to surface in current branch, how hard is it now to roll own custom modules?


P.S. If Discord let us host own servers I would do that, but from the little I've looked into it we can't. I do know its popular within game community.

**Comment by BrotherhoodOfHam** [comment]
> Have you at all looked at playing with the Vulkan spec, I see DX11 now?

I've tried vulkan/dx12 yet but I have done alot of reading on it. It's possible to add a different graphics API backend but it would likely mean refactoring the [abstraction layer](https://github.com/BrotherhoodOfHam/TSEngine/blob/master/modules/tsgraphics/inc/tsgraphics/Driver.h) some more.

>have any design decisions been made along the way with future cross platform ease of use

I've tried to avoid exposing platform specific headers publicly for this reason. Most of the work for making things platform-agnostic has gone into the graphics backend, everything else (input/windowing) could probably be handled by just using SDL or something.

Also cmake helps (and conan).

>how hard is it now to roll own custom modules?

A "module" really is just a static library in this case. [This](https://github.com/BrotherhoodOfHam/TSEngine/blob/2ea5e654901d667a108bc316b5f5f7529cac7478/modules/tscore/CMakeLists.txt) is an example, if you're familiar with cmake it's not that hard and completely optional all it does is [setup](https://github.com/BrotherhoodOfHam/TSEngine/blob/02c5df629179bba966e048b670d798398a4a6fac/cmake/AddModule.cmake) a few install rules and other useful things.

Add me on steam then if you prefer.
